app/Http/Controllers/UserController.php

```diff
@@ -61,4 +61,25 @@ public function update(UpdateUserRequest $request)
 
         return response($response, 200);
     }
+    public function getUsers(){
+        $users = User::all();
+        $email = $users->pluck('email');
+        $response = ['users' => $email];
+
+        return response($response, 200);
+    }
+    public function getUserData()
+    {
+        $user = Auth::user();
```

Comment:
incorrect, what is going on here? you just check that user is logged in and return error like: 'you haven't enough permissions'. You should add endpoints to 'auth' group and use $user->can method for permissions cheeck.